index.html

```diff
@@ -0,0 +1,96 @@
+<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
+<html xmlns="http://www.w3.org/1999/xhtml">
+<head>
+    <style>
+        html, body {height: 100%; margin: 5px; padding: 0;}
+        table {background-color: linen; font-family: "Verdana", Times, serif;}
+        th {background-color: #4CAF50;color: white;}
+        tr:nth-child(even) {background-color: #f2f2f2}
+        p {font-family: "Verdana", Times, serif;}
+    </style>
+    <link rel="stylesheet" type="text/css" href="mapStyle.css">
+	<meta name="viewport" content="width=device-width, initial-scale=1" />
+    <title>ArcTouch Code Challenge</title>
+    <script type="text/javascript" src="prototype.js"></script>
+    <script type="text/javascript" src="utils.js"></script>
+    <script type="text/javascript" src="map.js"></script>
+    <script type="text/javascript">
+        //<![CDATA[
+
+        function getPreviousRouteList() {
+            var street = getURLParameterByName("street");
+            if (street != null) {
+                document.getElementById("txtStreetName").value = street;
+                getRouteListByStreetName();
+            }
+        }
+
+        function getRouteListByStreetName() {
+            var txtStreetName = document.getElementById("txtStreetName").value.replace(/^\s+|\s+$/gm,'');

```

Comment:
I created getRoutesListAPIResponse() to separate the http request from the
rest. Code changed and committed. Thanks.

2016-03-02 17:08 GMT-03:00 André Luiz Cardoso <notifications@github.com>:

> In index.html
> <https://github.com/otaviofontana/arcTouchOtavioFontana/pull/1#discussion_r54782949>
> :
>
> > +    <script type="text/javascript" src="prototype.js"></script>
> > +    <script type="text/javascript" src="utils.js"></script>
> > +    <script type="text/javascript" src="map.js"></script>
> > +    <script type="text/javascript">
> > +        //<![CDATA[
> > +
> > +        function getPreviousRouteList() {
> > +            var street = getURLParameterByName("street");
> > +            if (street != null) {
> > +                document.getElementById("txtStreetName").value = street;
> > +                getRouteListByStreetName();
> > +            }
> > +        }
> > +
> > +        function getRouteListByStreetName() {
> > +            var txtStreetName = document.getElementById("txtStreetName").value.replace(/^\s+|\s+$/gm,'');
>
> This function has a lot of mixed responsibilities. It reads the input from
> the UI, validates it and does a network request. How would you do to fix
> that?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/otaviofontana/arcTouchOtavioFontana/pull/1/files#r54782949>
> .
>